examples/audio/main/bsp_audio_example.c

```diff
@@ -59,75 +51,22 @@ typedef struct __attribute__((packed))
     uint8_t data[];
 } dumb_wav_header_t;
 
-static esp_err_t spiffs_init(void)
-{
-    esp_err_t ret = ESP_OK;
-    ESP_LOGI(TAG, "Initializing SPIFFS");
-
-    const esp_vfs_spiffs_conf_t conf = {
-        .base_path = "/spiffs",
-        .partition_label = NULL,
-        .max_files = 5,
-        .format_if_mount_failed = false
-    };
-
-    /*!< Use settings defined above to initialize and mount SPIFFS filesystem. */
-    /*!< Note: esp_vfs_spiffs_register is an all-in-one convenience function. */
-    ret = esp_vfs_spiffs_register(&conf);
-
-    if (ret != ESP_OK) {
-        if (ret == ESP_FAIL) {
-            ESP_LOGE(TAG, "Failed to mount or format filesystem");
-        } else if (ret == ESP_ERR_NOT_FOUND) {
-            ESP_LOGE(TAG, "Failed to find SPIFFS partition");
-        } else {
-            ESP_LOGE(TAG, "Failed to initialize SPIFFS (%s)", esp_err_to_name(ret));
-        }
-
-        return ret;
-    }
-
-    size_t total = 0, used = 0;
-    ret = esp_spiffs_info(NULL, &total, &used);
-
-    if (ret != ESP_OK) {
-        ESP_LOGE(TAG, "Failed to get SPIFFS partition information (%s)", esp_err_to_name(ret));
-    } else {
-        ESP_LOGI(TAG, "Partition size: total: %d, used: %d", total, used);
-    }
-
-    return ret;
-}
-
 static void audio_task(void *arg)
 {
-    /* Create and configure ES8311 I2C driver */
-    es8311_handle_t es8311_dev = es8311_create(BSP_I2C_NUM, ES8311_ADDRRES_0);
-    const es8311_clock_config_t clk_cfg = BSP_ES8311_SCLK_CONFIG(SAMPLE_RATE);
-    es8311_init(es8311_dev, &clk_cfg, ES8311_RESOLUTION_16, ES8311_RESOLUTION_16);
-    es8311_voice_volume_set(es8311_dev, DEFAULT_VOLUME, NULL);
-
-    /* Microphone settings */
-    es7210_dev_handle_t es7210_handle = NULL;
-    const es7210_i2c_config_t es7210_i2c = BSP_ES7210_I2C_CONFIG(BSP_I2C_NUM);
-    ESP_ERROR_CHECK(es7210_new_codec(&es7210_i2c, &es7210_handle));
-    const es7210_codec_config_t codec_conf = BSP_ES7210_CONFIG(22050, I2S_MCLK_MULTIPLE_512);
-    ESP_ERROR_CHECK(es7210_config_codec(es7210_handle, &codec_conf));
-    ESP_ERROR_CHECK(es7210_config_volume(es7210_handle, 32));
-
-    /* Configure I2S peripheral and Power Amplifier */
-    bsp_audio_init(NULL, &i2s_tx_chan, &i2s_rx_chan);
-    bsp_audio_poweramp_enable(true);
+    esp_codec_dev_handle_t spk_codec_dev = bsp_audio_codec_speaker_init();
+    esp_codec_dev_handle_t mic_codec_dev = bsp_audio_codec_microphone_init();
+    assert(spk_codec_dev && mic_codec_dev);
```

Comment:
Can we change the example for using without mic and assert for mic can be only when we want to use it? It can help in future for boards without mic.